adafruit_led_animation/animation.py

```diff
@@ -0,0 +1,585 @@
+    """
+    Animate a sequence of one or more colors, cycling at the specified speed.
+
+    :param pixel_object: The initialised LED object.
+    :param int speed: Animation speed in seconds, e.g. ``0.1``.
+    :param colors: A list of colors to cycle through in ``(r, g, b)`` tuple, or ``0x000000`` hex
+                   format. Defaults to a rainbow color cycle.
+    """
+    def __init__(self, pixel_object, speed, colors=RAINBOW):
+        self.colors = colors
+        super(ColorCycle, self).__init__(pixel_object, speed, colors[0])
+        self._generator = self._color_generator()
+
+    def draw(self):
+        next(self._generator)
+        self.pixel_object.fill(self.color)
+        self.show()
+
+    def _color_generator(self):
+        index = 0
+        while True:
+            self._color = self.colors[index]
+            yield
+            index = (index + 1) % len(self.colors)
+
+
+class Blink(ColorCycle):
+    """
+    Blink a color on and off.
+
+    :param pixel_object: The initialised LED object.
+    :param int speed: Animation speed in seconds, e.g. ``0.1``.
+    :param color: Animation color in ``(r, g, b)`` tuple, or ``0x000000`` hex format.
+    """
+    def __init__(self, pixel_object, speed, color):
+        super(Blink, self).__init__(pixel_object, speed, [color, BLACK])
+
+    def _recompute_color(self, color):
+        self.colors = [color, BLACK]
+
+
+class Solid(ColorCycle):
+    """
+    A solid color.
+
+    :param pixel_object: The initialised LED object.
+    :param color: Animation color in ``(r, g, b)`` tuple, or ``0x000000`` hex format.
+    """
+    def __init__(self, pixel_object, color):
+        super(Solid, self).__init__(pixel_object, speed=1, colors=[color])
+
+    def _recompute_color(self, color):
+        self.colors = [color]
+
+
+class Comet(Animation):
+    """
+    A comet animation.
+
+    :param pixel_object: The initialised LED object.
+    :param int speed: Animation speed in seconds, e.g. ``0.1``.
+    :param color: Animation color in ``(r, g, b)`` tuple, or ``0x000000`` hex format.
+    :param int tail_length: The length of the comet. Defaults to 10. Cannot exceed the number of
+                            pixels present in the pixel object, e.g. if the strip is 30 pixels
+                            long, the ``tail_length`` cannot exceed 30 pixels.
+    :param bool reverse: Animates the comet in the reverse order. Defaults to ``False``.
+    :param bool bounce: Comet will bounce back and forth. Defaults to ``True``.
+    """
+    # pylint: disable=too-many-arguments
+    def __init__(self, pixel_object, speed, color, tail_length=10, reverse=False, bounce=False):
+        self._tail_length = tail_length + 1
+        self._color_step = 0.9 / tail_length
+        self._color_offset = 0.1
+        self._comet_colors = None
+        self._reverse_comet_colors = None
+        self.reverse = reverse
+        self.bounce = bounce
+        super(Comet, self).__init__(pixel_object, speed, color)
```

Comment:
nit: it's unusual to see the superclass `__init__` so far down. I've generally seen this done as soon as possible, like in the very first statement in the `__init__` method.

Reply:
I usually do it first, but in this case, when I did this, the color setup work needed doing before the rest of super is run.  I'll check if that's still true.  If not, a comment would be useful.

Reply:
Comments added and init for both the base class and this class reorganized slightly.